src/cryptonote_core/blockchain.cpp

```diff
@@ -3648,11 +3648,25 @@ uint64_t Blockchain::get_dynamic_base_fee(uint64_t block_reward, size_t median_b
   if (version >= HF_VERSION_PER_BYTE_FEE)
   {
     lo = mul128(block_reward, DYNAMIC_FEE_REFERENCE_TRANSACTION_WEIGHT, &hi);
-    div128_64(hi, lo, min_block_weight, &hi, &lo, NULL, NULL);
     div128_64(hi, lo, median_block_weight, &hi, &lo, NULL, NULL);
-    assert(hi == 0);
-    lo /= 5;
-    return lo;
+    if (version >= HF_VERSION_2021_SCALING)
+    {
+      // min_fee_per_byte = round_up( 0.95 * block_reward * ref_weight / (fee_median^2) )
+      // note: since hardfork HF_VERSION_2021_SCALING, fee_median (a.k.a. median_block_weight) equals effective long term median
+      div128_64(hi, lo, median_block_weight, &hi, &lo, NULL, NULL);
+      assert(hi == 0);
+      lo *= CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5;
+      lo /= median_block_weight;
+      lo -= lo / 20;
+      return lo;
+    }
```

Comment:
```suggestion
    if (version >= HF_VERSION_2021_SCALING)
    {
      // min_fee_per_byte = round_up( 0.95 * block_reward * ref_weight / (fee_median^2) )
      // note: since hardfork HF_VERSION_2021_SCALING, fee_median (a.k.a. median_block_weight) equals effective long term median
      div128_64(hi, lo, median_block_weight, &hi, &lo, NULL, NULL);
      assert(hi == 0);
      lo -= lo / 20;
      return lo;
    }
```

I am confused why you want these two lines in this function. Am I missing something in the fee doc?

Reply:
Are they not in the doc ? I assume they are if I ended up with that code. I'll double check later and remove if I don't find them and it still works with them out.

Reply:
I looked and I cannot see that. I have no idea why I put those in. Removed. I'll push once tests have run.